src/features/island/buildings/components/building/kitchen/Kitchen.tsx

```diff
@@ -30,6 +34,13 @@ export const Kitchen: React.FC<Props> = ({
 }) => {
   const [showModal, setShowModal] = useState(false);
 
+  const { gameService } = useContext(Context);
+  const [gameState] = useActor(gameService);
```

Comment:
For optimisation, are we able to use `useSelector` here?

It may be a too much to cause a re-render whenever any game state changes, instead we can listen and react only to Bumpkin experience change.

Reply:
I will include an example of how we can achieve this 👍 One minute 🤓

Reply:
```
const _level = (state: MachineState) =>
  getBumpkinLevel(state.context.state.bumpkin?.experience ?? 0);

export const Kitchen: React.FC<Props> = ({
  buildingId,
  crafting,
  idle,
  ready,
  name,
  craftingService,
  isBuilt,
  onRemove,
}) => {
  const [showModal, setShowModal] = useState(false);

  const { gameService } = useContext(Context);

  const bumpkinLevel = useSelector(gameService, _level);
  const is_kitchen_enabled = isBuildingEnabled(bumpkinLevel, "Kitchen");
  ```

Reply:
> For optimisation, are we able to use `useSelector` here?
> 
> It may be a too much to cause a re-render whenever any game state changes, instead we can listen and react only to Bumpkin level changes.

Updated